### PR TITLE
[BEAM-1320] Fix documentation in the sdk and fail for new errors

### PR DESCRIPTION
**Pull request by sb2nov**
R: @aaltay PTAL

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

File: sdks/python/generate_pydoc.sh [hunk 1]
```diff
@@ -59,8 +62,11 @@ EOF
...
 python $(type -p sphinx-build) -q target/docs/source target/docs/_build -c target/docs/source \
-    -w "/tmp/sphinx-build.warnings.log"
+    -w "target/docs/sphinx-build.warnings.log"
 
 # Message is useful only when this script is run locally.  In a remote
 # test environment, this path will be removed when the test completes.
 echo "Browse to file://$PWD/target/docs/_build/index.html"
+
+# Fail if there are errors in docs
+! grep -q "ERROR:" target/docs/sphinx-build.warnings.log
```
Comment (aaltay) [inline sdks/python/generate_pydoc.sh]: Should this fail for WARNINGS too?

Comment (sb2nov) [inline sdks/python/generate_pydoc.sh]: Yes, we still have some more warnings that we need to squash.

Comment (aaltay) [inline sdks/python/generate_pydoc.sh]: Sounds good, we can keep tracking them on the issue. This is a good change by itself.

**Comment by aaltay** [comment]
LGTM

**Comment by sb2nov** [comment]
It was a lint issue. Fixed and pushed

File: sdks/python/apache_beam/transforms/core.py [hunk 2]
```diff
@@ -915,10 +914,10 @@ class CombinePerKey(PTransformWithSideInputs):
   Args:
     pcoll: input pcollection.
     fn: instance of CombineFn to apply to all values under the same key in
-      pcoll, or a callable whose signature is f(iterable, *args, **kwargs)
+      pcoll, or a callable whose signature is f(iterable, \*args, \**kwargs)
       (e.g., sum, max).
-    *args: arguments and side inputs, passed directly to the CombineFn.
-    **kwargs: arguments and side inputs, passed directly to the CombineFn.
+    \*args: arguments and side inputs, passed directly to the CombineFn.
```
Comment (aaltay) [inline sdks/python/apache_beam/transforms/core.py]: Why this is escaped here `\*` and not in L145?

Comment (sb2nov) [inline sdks/python/apache_beam/transforms/core.py]: Thanks for catching. It was only needed in the line above this and not here. Fixed now.

**Comment by aaltay** [comment]
LGTM

**Pull request closed without merging**